ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriteForRowtime.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.parser.rewrite;
+
+import io.confluent.ksql.parser.tree.BetweenPredicate;
+import io.confluent.ksql.parser.tree.ComparisonExpression;
+import io.confluent.ksql.parser.tree.Expression;
+import io.confluent.ksql.parser.tree.FunctionCall;
+import io.confluent.ksql.parser.tree.Node;
+import io.confluent.ksql.parser.tree.QualifiedName;
+import io.confluent.ksql.parser.tree.StringLiteral;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class StatementRewriteForRowtime {
+  private final Expression expression;
+
+  public StatementRewriteForRowtime(final Expression expression) {
+    this.expression = Objects.requireNonNull(expression, "expression");
+  }
+
+  public static boolean requiresRewrite(final Expression expression) {
+    return expression.toString().contains("ROWTIME");
```

Comment:
This will match any expression with `ROWTIME` in it. For instance, it will match the following expression where you should not indeed rewrite the expression:
`TIMESTAMPTOSTRING(ROWTIME, 'yyyy-MM-dd HH:mm:ss') = '2019-08-08 12:12:12'`.
You need to change this such that it only rewrites the cases you have mentioned in the docs.

Reply:
This particular statement remained the same, but the rewriter itself has been updated to only rewrite timestrings when one side of the comparison is exactly `ROWTIME`.